spec/do_username_spec.rb

```diff
@@ -0,0 +1,81 @@
+RSpec.describe DOUsername do
+  describe '.generate' do
+    subject { described_class }
+
+    # Please update this spec if module constants are updated
+    # see more at: https://github.com/MattIPv4/do_username/pull/8
+    it 'responds with different usernames based on srand' do
+      [
+        'HappyElectricBlueBoat', 'DriftingAquaShark', 'MassiveTurquoiseWhale'
+      ].each_with_index do |username, i|
+        srand i
+
+        expect(subject.generate).to eq(username)
+      end
+    end
+
+    it 'responds with string' do
+      expect(subject.generate).to be_a String
+    end
+
+    context 'with noun part' do
+      before { stub_const('DOUsername::SEA_LIST', ['walrus']) }
+
+      it 'ends with a sea object or a creature' do
+        expect(subject.generate).to end_with('Walrus')
+      end
+    end
+
+    context 'with descriptor part' do
+      before do
+        stub_const('DOUsername::DESCRIPTORS', ['cute'])
```

Comment:
```suggestion
        stub_const('DOUsername::ADJECTIVE_DESCRIPTORS', ['cute'])
```
I think this needs to stub `ADJECTIVE_DESCRIPTORS` now?

Reply:
Actually no, because `DESCRIPTORS` is an array generated based on `SIZE_DESCRIPTORS + ADJECTIVE_DESCRIPTORS` so, stubbing descriptors directly should be fine :)

Reply:
We're stubbing `SIZE_DESCRIPTORS` on the line below though, so either we should just stub `DESCRIPTORS`, or both `SIZE_DESCRIPTORS` & `ADJECTIVE_DESCRIPTORS` so that it makes a bit more sense?

Reply:
Oh, yes! I'll change it!

Reply:
I've tried stubbing both `adjectives` and `sizes` (without the `DESCRIPTORS`) and not work because the execution order, so stubbing just the `DESCRIPTORS` worked!